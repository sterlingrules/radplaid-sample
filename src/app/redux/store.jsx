import defaults from 'lodash/defaults'
import { routerReducer } from 'react-router-redux'
import { persistStore, persistReducer, persistCombineReducers } from 'redux-persist'
import { applyMiddleware, createStore, combineReducers, compose } from 'redux'
// import { CookieStorage } from 'redux-persist-cookie-storage'
import { DEBUG } from './../constants.jsx'
import { DEFAULT_APP, DEFAULT_SHOWS } from './../constants.computed.jsx'
import storage from 'redux-persist/lib/storage'
import createLogger from 'redux-logger'
import Cookies from 'js-cookie'
import thunk from 'redux-thunk'

import app from './reducers/app.jsx'
import user from './reducers/user.jsx'
import shows from './reducers/shows.jsx'
import player from './reducers/player.jsx'
import orders from './reducers/orders.jsx'
import voter from './reducers/voter.jsx'

const logger = createLogger()
// const composeEnhancers = (typeof window !== 'undefined') ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ : compose

const INITIAL_STORE = {
	app: DEFAULT_APP,
	shows: DEFAULT_SHOWS,
}

const persistConfig = {
	key: 'root',
	storage,
	stateReconciler: (inboundState, originalState, reducedState) => {
		if (typeof window === 'undefined' || !window.REDUX_DATA) {
			return reducedState
		}

		reducedState.user = defaults(window.REDUX_DATA.user, reducedState.user)

		return reducedState
	},
	whitelist: [
		'user'
	],
	blacklist: [
		'app',
		'shows',
		'player',
		'voter',
		'routing',
		'orders',
	]
}

const appPersistConfig = {
	key: 'app',
	storage,
	stateReconciler: (inboundState, originalState, reducedState) => {
		if (typeof window === 'undefined' || !window.REDUX_DATA) {
			return reducedState
		}

		return defaults(window.REDUX_DATA.app, reducedState)
	},
	blacklist: [
		'initialized',
		'loggingIn',
		'session',
		'connection',
		'isBrowser',
		'progress',
		'progressStatus',

		'messageId',
		'message',
		'messageTitle',
		'messageType',
		'messageAction',
		'messageSticky',
		'messageClear',
		'messageChildren',

		'isSearching',
		// 'searchSort',
		'searchQuery',
		'searchVenue',
		'searchCost',
		'searchTag',
		'searchLocation',
		'searchDateFrom',
		'searchDateTo',
	],
}

const showsPersistConfig = {
	key: 'shows',
	storage,
	stateReconciler: (inboundState, originalState, reducedState) => {
		if (typeof window === 'undefined' || !window.REDUX_DATA) {
			return reducedState
		}

		return defaults(window.REDUX_DATA.shows, reducedState)
	},
	blacklist: [
		'activeShow',
	],
}

export const persistedReducer = persistReducer(
	persistConfig,
	combineReducers({
		app: persistReducer(appPersistConfig, app),
		shows: persistReducer(showsPersistConfig, shows),
		player,
		user,
		voter,
		orders,
		routing: routerReducer,
	}),
	(typeof window !== 'undefined') ? window.REDUX_DATA : INITIAL_STORE,
)

console.log('DEBUG ', DEBUG)
console.log('process.env.NODE_ENV ', process.env.NODE_ENV)

export const store = createStore(
	persistedReducer,
	compose(
		DEBUG ? applyMiddleware(thunk, logger) : applyMiddleware(thunk)
	)
)

export const persistor = persistStore(store)
