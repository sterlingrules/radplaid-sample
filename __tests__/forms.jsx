require('dotenv').config()

import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import store from './../src/app/redux/store.jsx'
import InputScenes from './../src/app/forms/input-scenes.jsx'
import InputEmail from './../src/app/forms/input-email.jsx'
import Search from './../src/app/common/header/search.jsx'

configure({ adapter: new Adapter() })

jest.useFakeTimers()

describe('InputEmail', () => {
	const InputEmailName = 'input-email'

	it('renders correctly', () => {
		const InputEmailComponent = shallow(<InputEmail name={InputEmailName} />)
		expect(InputEmailComponent).toMatchSnapshot()
	})

	it('returns valid email', () => {
		const onInputChangeMock = jest.fn()
		const InputEmailComponent = shallow(
			<InputEmail
				userId="newuser"
				name={InputEmailName}
				onInputChange={onInputChangeMock} />
		)

		let event = {
			target: {
				name: InputEmailName,
				value: 'yo@sterlingrul.es'
			}
		}

		InputEmailComponent.simulate('change', event)

		expect(onInputChangeMock).toHaveBeenCalledTimes(1)
		expect(onInputChangeMock).toHaveBeenCalledWith({
			target: {
				name: InputEmailName,
				value: 'yo@sterlingrul.es',
				valid: true,
				validType: 'format',
				validClass: 'form--valid',
				message: 'Valid email please. We never spam.'
			}
		})
	})

	it('handles malformed email', () => {
		const onInputChangeMock = jest.fn()
		const InputEmailComponent = shallow(
			<InputEmail
				userId="newuser"
				name={InputEmailName}
				onInputChange={onInputChangeMock} />
		)

		let event = {
			target: {
				name: InputEmailName,
				value: 'yo@sterlingrul'
			}
		}

		InputEmailComponent.simulate('change', event)

		expect(onInputChangeMock).toHaveBeenCalledTimes(1)
		expect(onInputChangeMock).toHaveBeenCalledWith({
			target: {
				name: InputEmailName,
				value: 'yo@sterlingrul',
				valid: false,
				validType: 'format',
				validClass: 'form--invalid',
				message: 'Valid email please. We never spam.'
			}
		})
	})

	it('handles duplicate email', () => {
		const onInputChangeMock = jest.fn()
		const InputEmailComponent = shallow(
			<InputEmail
				userId="newuser"
				name={InputEmailName}
				validateDupes={true}
				onInputChange={onInputChangeMock} />
		)

		let event = {
			target: {
				name: InputEmailName,
				value: 'hello@getradplaid.com'
			}
		}

		InputEmailComponent.simulate('change', event)

		setTimeout(() => {
			expect(onInputChangeMock).toHaveBeenCalledTimes(2)
			expect(onInputChangeMock).toHaveBeenLastCalledWith({
				target: {
					name: InputEmailName,
					value: 'hello@getradplaid.com',
					valid: false,
					validType: 'duplicate',
					validClass: 'form--invalid',
					message: 'Yikes, that email is already taken.'
				}
			})
		}, 1000)
	})
})

describe('HeaderSearch', () => {
	describe('large viewport', () => {
		it('renders correctly', () => {
			const SearchComponent = shallow(<Search viewportName="large" />)
			expect(SearchComponent).toMatchSnapshot()
		})
	})

	describe('small viewport', () => {
		it('renders correctly', () => {
			const SearchComponent = shallow(<Search viewportName="small" />)
			expect(SearchComponent).toMatchSnapshot()
		})
	})
})

describe('InputScenes', () => {
	it('renders correctly', () => {
		expect(shallow(<InputScenes name="input-scenes" />)).toMatchSnapshot()
	})
})
