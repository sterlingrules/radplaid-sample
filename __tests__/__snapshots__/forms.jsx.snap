// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderSearch large viewport renders correctly 1`] = `
<Route>
  <Component />
</Route>
`;

exports[`HeaderSearch small viewport renders correctly 1`] = `
<Route>
  <Component />
</Route>
`;

exports[`InputEmail renders correctly 1`] = `
<input
  autoCapitalize="none"
  autoComplete="off"
  autoCorrect="off"
  className="form-input text-right"
  name="email"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  placeholder=""
  spellCheck="false"
  type="email"
  value=""
/>
`;

exports[`InputScenes renders correctly 1`] = `
<span
  className=""
>
  <Select
    options={Array []}
    settings={
      Object {
        "closeMenuOnScroll": false,
        "closeMenuOnSelect": true,
        "isMulti": true,
        "isSearchable": true,
        "name": "input-scenes",
        "onChange": [Function],
        "onInputChange": [Function],
        "placeholder": "Select your Music Scenes",
        "value": undefined,
      }
    }
    type="simple"
  />
</span>
`;
